File: spec/javascripts/unit/Filter_spec.js
describe("Filters", function() {
  describe("#init", function() {
  });

  describe("#sendPhotoUrl", function() {
    beforeEach(function() {
      var api = { sendPhotoUrl: function()  { return {  complete: function(callback) { callback("woo"); } } } }
      spyOn(Filters, 'renderHTML');
      spyOn(Filters, 'createPreview');
      Filters.sendPhotoUrl('http://image.com/img.png', api);
    });

    describe("When it completes successully", function() {
      it("Renders the html with the api response", function() {
        expect(Filters.renderPartial).toHaveBeenCalledWith("woo");
      });

      it("creates the preview", function() {
        expect(Filters.createPreview).toHaveBeenCalled();
      });
    });
  });
});

File: spec/javascripts/unit/Filter_spec.rb
describe("Filters", function(){

  describe("#init", function() {
    before("all", function() {
      spyOn(Filters, 'registerFilters');
    });

    it("registers filters", function() {
      Filters.init()

      expect(Filters.registerFilters).toHaveBeenCalled();
    })
  });

});